x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/create_threat_signal.ts

```diff
@@ -51,68 +49,67 @@ export const createThreatSignal = async ({
   name,
   currentThreatList,
   currentResult,
-}: CreateThreatSignalOptions): Promise<{
-  threatList: SearchResponse<ThreatListItem>;
-  results: SearchAfterAndBulkCreateReturnType;
-}> => {
-  const threatFilter = buildThreatMappingFilter({
-    threatMapping,
-    threatList: currentThreatList,
-  });
-
-  const esFilter = await getFilter({
-    type,
-    filters: [...filters, threatFilter],
-    language,
-    query,
-    savedId,
-    services,
-    index: inputIndex,
-    lists: exceptionItems,
-  });
-
-  const newResult = await searchAfterAndBulkCreate({
-    gap,
-    previousStartedAt,
-    listClient,
-    exceptionsList: exceptionItems,
-    ruleParams: params,
-    services,
-    logger,
-    eventsTelemetry,
-    id: alertId,
-    inputIndexPattern: inputIndex,
-    signalsIndex: outputIndex,
-    filter: esFilter,
-    actions,
-    name,
-    createdBy,
-    createdAt,
-    updatedBy,
-    updatedAt,
-    interval,
-    enabled,
-    pageSize: searchAfterSize,
-    refresh,
-    tags,
-    throttle,
-    buildRuleMessage,
-  });
-
-  const results = combineResults(currentResult, newResult);
-  const searchAfter = currentThreatList.hits.hits[currentThreatList.hits.hits.length - 1].sort;
-
+}: CreateThreatSignalOptions): Promise<ThreatSignalResults> => {
   const threatList = await getThreatList({
```

Comment:
nit: it's a little confusing that `threatList` is now created at the beginning of the function, but is only used as the return value at the end

Reply:
Yeah, I thought so too. I did it to reduce the amount of copy code otherwise.